inmem.go

```diff
@@ -154,12 +154,16 @@ func NewInmemSink(interval, retain time.Duration) *InmemSink {
 }
 
 func (i *InmemSink) SetGauge(key []string, val float32) {
-	k := i.flattenKey(key)
+	i.SetGaugeWithLabels(key, val, nil)
+}
+
+func (i *InmemSink) SetGaugeWithLabels(key []string, val float32, labels []Label) {
+	k := i.flattenKeyLabels(key, labels)
 	intv := i.getInterval()
 
 	intv.Lock()
 	defer intv.Unlock()
-	intv.Gauges[k] = val
+	intv.Gauges[k] = GaugeValue{Name: i.flattenKey(key), Value: val, Labels: labels}
```

Comment:
Since there's an implicit `flattenKey()` inside `flattenKeyLabels()` and this is a pretty hot path, how about:

`k, name := i.flattenKeyLabels`

And then you can just plug name into the structure?